### DISCUSSION TITLE
Update password through the ui?

### DISCUSSION THREAD

**Discussion by tobias-z** [argoproj/argo-cd#19011] [Q&A]
Hi there,
I am sorry if this question has already been answered somewhere else. I have tried to find a question about it but couldn't.

Is it correct that the only way to update a users password is through the cli?
I find it really weird that all users must use the argocd cli in order to update their users password. Admins are usually quite good with clis but it is simply a fact that sometimes we give logins to people who are either not great at cli's or have never touched one before. This is the case with some people who simply need to be able to login and see the status of some applications deployed or something like that.

If this is the case I would very much like a feature of letting users update their own password using the ui, so we dont have to tell them: `just download this cli, login and then run this command`

**Comment by ChristianCiach** [comment]
I believe Argo-CD isn't really designed for complex user management. User management should probably better be done externally. I guess this is why Argo-CD bundles Dex, so you can use one of [Dex' Connectors ](https://dexidp.io/docs/connectors/) to connect to your external user management service.

For example, we use Keycloak for user management and use Argo-CD's native OIDC support to [connect Argo-CD to our Keycloak instance](https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/keycloak/).
